plonky2x/src/frontend/eth/beacon/builder.rs

```diff
@@ -1,173 +1,568 @@
     use crate::frontend::vars::Bytes32Variable;
-    use crate::prelude::Variable;
-    use crate::utils::bytes32;
     use crate::utils::eth::beacon::BeaconClient;
+    use crate::utils::{bytes, bytes32};
+
+    type F = GoldilocksField;
+    type C = PoseidonGoldilocksConfig;
+    const D: usize = 2;
 
     #[test]
     #[cfg_attr(feature = "ci", ignore)]
-    fn test_get_validator_generator() {
-        env_logger::init();
+    fn test_beacon_get_validators() {
+        env_logger::try_init().unwrap_or_default();
         dotenv::dotenv().ok();
 
-        type F = GoldilocksField;
-        type C = PoseidonGoldilocksConfig;
-        const D: usize = 2;
+        let consensus_rpc = env::var("CONSENSUS_RPC_1").unwrap();
+        let client = BeaconClient::new(consensus_rpc);
+        let latest_block_root = client.get_finalized_block_root_sync().unwrap();
+
+        let mut builder = CircuitBuilder::<F, D>::new();
+        builder.set_beacon_client(client);
+
+        let block_root = builder.constant::<Bytes32Variable>(bytes32!(latest_block_root));
+        let _validators = builder.beacon_get_validators(block_root);
+        // builder.watch(&validators, "validators");
+
+        let circuit = builder.build::<C>();
+        let input = circuit.input();
+        let (proof, output) = circuit.prove(&input);
+        circuit.verify(&proof, &input, &output);
+    }
 
-        let consensus_rpc = env::var("CONSENSUS_RPC_URL").unwrap();
+    #[test]
+    #[cfg_attr(feature = "ci", ignore)]
+    fn test_beacon_get_validator() {
+        env_logger::try_init().unwrap_or_default();
+        dotenv::dotenv().ok();
+
+        let consensus_rpc = env::var("CONSENSUS_RPC_1").unwrap();
         let client = BeaconClient::new(consensus_rpc);
+        let latest_block_root = client.get_finalized_block_root_sync().unwrap();
 
         let mut builder = CircuitBuilder::<F, D>::new();
+        builder.set_beacon_client(client);
+
+        let block_root = builder.constant::<Bytes32Variable>(bytes32!(latest_block_root));
+        let validators = builder.beacon_get_validators(block_root);
+        let index = builder.constant::<U64Variable>(0.into());
+        let validator = builder.beacon_get_validator(validators, index);
+        let expected_validator_pubkey = builder.constant::<BLSPubkeyVariable>(bytes!(
+            "0x933ad9491b62059dd065b560d256d8957a8c402cc6e8d8ee7290ae11e8f7329267a8811c397529dac52ae1342ba58c95"
+        ));
+        builder.assert_is_equal(validator.pubkey, expected_validator_pubkey);
+
+        let circuit = builder.build::<C>();
+        let input = circuit.input();
+        let (proof, output) = circuit.prove(&input);
+        circuit.verify(&proof, &input, &output);
+    }
+
+    #[test]
+    #[cfg_attr(feature = "ci", ignore)]
+    fn test_beacon_get_validator_const() {
+        env_logger::try_init().unwrap_or_default();
+        dotenv::dotenv().ok();
+
+        let consensus_rpc = env::var("CONSENSUS_RPC_1").unwrap();
```

Comment:
Please add a `.env.example`, otherwise it's difficult to track necessary .env variables (considering we change their names often)